### PR TITLE
Port `log4j-to-slf4j` changes from `2.x`

### PR DESCRIPTION
**Pull request by vy**

### CODE REVIEW

File: log4j-to-slf4j/src/test/java/org/apache/logging/slf4j/LogBuilderTest.java [hunk 1]
```diff
@@ -31,31 +30,29 @@
...
+@LoggerContextSource
 public class LogBuilderTest {
 
-    private static final String CONFIG = "target/test-classes/logback-turbofilter.xml";
     private static final CharSequence CHAR_SEQUENCE = "CharSequence";
     private static final String STRING = "String";
     private static final Message MESSAGE = new SimpleMessage();
     private static final Object[] P = {"p0", "p1", "p2", "p3", "p4", "p5", "p6", "p7", "p8", "p9"};
-    private static final Object OBJECT = "Object";
+    private static Object OBJECT = "Object";
```
[the remark this answers is not in the document]
Comment (vy) [reply log4j-to-slf4j/src/test/java/org/apache/logging/slf4j/LogBuilderTest.java:46]: This is not possible:
```
java.lang.UnsupportedOperationException: java.lang.IllegalAccessException: Can not set static final java.lang.Object field org.apache.logging.slf4j.LogBuilderTest.OBJECT to org.apache.logging.log4j.test.junit.TestPropertySource$JUnitTestProperties
	at org.apache.logging.log4j.util.ReflectionUtil.setFieldValue(ReflectionUtil.java:130)
	at org.apache.logging.log4j.util.ReflectionUtil.setStaticFieldValue(ReflectionUtil.java:143)
	at org.apache.logging.log4j.test.junit.TestPropertyResolver.lambda$beforeAll$5(TestPropertyResolver.java:64)
	at java.base/java.util.ArrayList.forEach(ArrayList.java:1541)
	at java.base/java.util.Collections$UnmodifiableCollection.forEach(Collections.java:1085)
	at org.apache.logging.log4j.test.junit.TestPropertyResolver.beforeAll(TestPropertyResolver.java:64)
	at java.base/java.util.ArrayList.forEach(ArrayList.java:1541)
Caused by: java.lang.IllegalAccessException: Can not set static final java.lang.Object field org.apache.logging.slf4j.LogBuilderTest.OBJECT to org.apache.logging.log4j.test.junit.TestPropertySource$JUnitTestProperties
	at java.base/java.lang.reflect.Field.set(Field.java:780)
	at org.apache.logging.log4j.util.ReflectionUtil.setFieldValue(ReflectionUtil.java:128)
	... 6 more
```

File: log4j-to-slf4j/src/test/java/org/apache/logging/slf4j/LoggerTest.java [hunk 2]
```diff
@@ -33,30 +33,28 @@
...
+@UsingStatusListener
+@LoggerContextSource
 public class LoggerTest {
 
-    private static final String CONFIG = "target/test-classes/logback-slf4j.xml";
-
-    @ClassRule
-    public static final LoggerContextRule CTX = new LoggerContextRule(CONFIG);
-
+    private static Object OBJ = new Object();
```
[the remark this answers is not in the document]
Comment (vy) [reply log4j-to-slf4j/src/test/java/org/apache/logging/slf4j/LoggerTest.java:45]: This is not possible due to `IllegalAccessException` I shared above.

File: log4j-to-slf4j/src/test/java/org/apache/logging/slf4j/LoggerTest.java [hunk 3]
```diff
@@ -73,7 +71,7 @@ public void basicFlow() {
 
     @Test
     public void simpleFlow() {
-        logger.traceEntry(CONFIG);
+        logger.trace(OBJ);
```
[the remark this answers is not in the document]
Comment (vy) [reply log4j-to-slf4j/src/test/java/org/apache/logging/slf4j/LoggerTest.java:74]: There is no `Logger#traceEntry(Object)` method. What do you mean?

**Comment by vy** [comment]
> This looks good to me but we might require an SLF4J/Logback upgrade:
> 
> * bump SLF4J to `2.0.9` (like `2.x`),
> * bump Logback to `1.4.11` (`2.x` uses the 1.3.x branch, since the 1.4.x branch requires Java 11).

I will address this while sync'ing `log4j-parent`. I would like to proceed one module at a time.

File: log4j-to-slf4j/src/test/java/org/apache/logging/slf4j/LogBuilderTest.java (hunk 1 above)
[the remark this answers is not in the document]
Comment (ppkarwasz) [reply log4j-to-slf4j/src/test/java/org/apache/logging/slf4j/LogBuilderTest.java:46]: It's a bug in `TestPropertyResolver`. I ported the fix to `main`.

File: log4j-to-slf4j/src/test/java/org/apache/logging/slf4j/LoggerTest.java (hunk 3 above)
[the remark this answers is not in the document]
Comment (ppkarwasz) [reply log4j-to-slf4j/src/test/java/org/apache/logging/slf4j/LoggerTest.java:74]: Yes, I noticed... ;-)

We don't break `log4j-api`, but we break `log4j-api` (or the methods were added after `main` and `2.x` diverged).

I replaced it with `logger.traceEntry(STRING)` in `main`.

**Comment by ppkarwasz** [comment]
> > This looks good to me but we might require an SLF4J/Logback upgrade:
> > 
> > * bump SLF4J to `2.0.9` (like `2.x`),
> > * bump Logback to `1.4.11` (`2.x` uses the 1.3.x branch, since the 1.4.x branch requires Java 11).
> 
> I will address this while sync'ing `log4j-parent`. I would like to proceed one module at a time.

I cherry picked the "Bump SLF4J to version 2.0.9" commit from `2.x`. It should be OK now.

Logback was bumped to `1.3.11` only, because `1.4.11` has a broken OSGi descriptor.

**Pull request merged**